autopush/db.py

```diff
@@ -537,25 +537,42 @@ def __init__(self, tablename, boto_resource=None,
     def table_status(self):
         return self.table.table_status
 
+    @track_provisioned
+    def touch_record(self, uaid, ttl=None):
+        # type: (str, int) -> None
+        if ttl is None:
+            ttl = self._max_ttl
+        self.table.update_item(
+            Key={
+                'uaid': hasher(uaid),
+                'chidmessageid': ' ',
+            },
+            # `list_append(chids, :channel_id)` errors if chids doesn't exist
+            UpdateExpression='SET expiry = :expiry',
+            ExpressionAttributeValues={
+                ":expiry": _expiry(ttl)
+            },
+        )
+
     @track_provisioned
     def register_channel(self, uaid, channel_id, ttl=None):
         # type: (str, str, int) -> bool
         """Register a channel for a given uaid"""
         # Generate our update expression
         if ttl is None:
             ttl = self._max_ttl
-        expr_values = {
-            ":channel_id": set([normalize_id(channel_id)]),
-            ":expiry": _expiry(ttl)
-        }
+        # Create/Append to list of channel_ids
         self.table.update_item(
             Key={
                 'uaid': hasher(uaid),
                 'chidmessageid': ' ',
             },
-            UpdateExpression='ADD chids :channel_id, expiry :expiry',
-            ExpressionAttributeValues=expr_values,
+            UpdateExpression='ADD chids :channel_id',
+            ExpressionAttributeValues={
+                ":channel_id": set([normalize_id(channel_id)]),
+            },
         )
+        self.touch_record(uaid)
```

Comment:
The chids and expiry can be changed in the same command, the Rust version does this, the update expression looks like this:
```rust
            update_expression: Some("ADD chids :channel_id SET expiry=:expiry".to_string()),
```